bridge-fungible-token/src/utils.sw

```diff
@@ -19,14 +28,56 @@ const GTF_INPUT_MESSAGE_DATA = 0x11E;
 const GTF_INPUT_MESSAGE_SENDER = 0x115;
 const GTF_INPUT_MESSAGE_RECIPIENT = 0x116;
 
+fn bn_mul(bn: U256, factor: u64) -> (b256, u64) {
+    disable_panic_on_overflow();
+    let product_and_overflow = asm(bn: __addr_of(bn), factor: factor, carry, new_carry, value, current_product, product_with_carry, prod_and_ovf) {
+        // Run multiplication on the lower 64bit word
+        lw value bn i3; // load the word in (bn + 3 words) into v
+        mul current_product value factor; // multiply v * d and save result in v
+        move new_carry of; // record the carry
+        move prod_and_ovf sp;
+        cfei i40;
```

Comment:
Why allocate the memory in assembly now when previously you were trying to avoid using assembly when possible (like with manipulating the panic and overflow flags)? 

Also, side note, all of the comments now need to reference the new names you gave the registers.

Reply:
Will update the comments now.